nrf52840-hal/memory.x

```diff
@@ -0,0 +1,23 @@
+MEMORY
+{
+  /* NOTE K = KiBi = 1024 bytes */
+  /* TODO Adjust these memory regions to match your device memory layout */
+  FLASH : ORIGIN = 0x00000000, LENGTH = 950K
```

Comment:
These seem a little smaller than I was expecting; are these adjusted for the softdevice?
https://github.com/particle-iot/nrf5_sdk/blob/6177fb3797962aaa091e04fa363bfb93a55eb7ef/modules/nrfx/mdk/nrf52840_xxaa.ld#L8

the memory.x in the hal should probably not factor in the softdevice and that should probably be something that is handled in the "application" crate.

Reply:
No, actually a good catch. When I first tried to get my dev board working I had a real mare of a time. Thinking back I think I reduced the sizes to try and prevent this weird issue I was having, I can't for the life of me remember my rationale. I'll put it back and remove the TODO.